src/app/aliases/aliases.editor.modal.ts

```diff
@@ -0,0 +1,263 @@
+// --------- BEGIN RUNBOX LICENSE ---------
+// Copyright (C) 2016-2018 Runbox Solutions AS (runbox.com).
+// 
+// This file is part of Runbox 7.
+// 
+// Runbox 7 is free software: You can redistribute it and/or modify it
+// under the terms of the GNU General Public License as published by the
+// Free Software Foundation, either version 3 of the License, or (at your
+// option) any later version.
+// 
+// Runbox 7 is distributed in the hope that it will be useful, but
+// WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// General Public License for more details.
+// 
+// You should have received a copy of the GNU General Public License
+// along with Runbox 7. If not, see <https://www.gnu.org/licenses/>.
+// ---------- END RUNBOX LICENSE ----------
+import { timeout } from 'rxjs/operators';
+import { 
+  SecurityContext, 
+  Component, 
+  Input, 
+  Output, 
+  EventEmitter, 
+  NgZone, 
+  ViewChild, 
+  Inject,
+  AfterViewInit
+} from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { FormsModule } from '@angular/forms';
+import { HttpClient, HttpHeaders } from '@angular/common/http'
+import {
+  MatCardModule,
+  MatCheckboxModule,
+  MatDialogModule,
+  MatInputModule,
+  MatListModule,
+  MatPaginatorModule,
+  MatProgressBarModule,
+  MatProgressSpinnerModule,
+  MatSnackBarModule,
+  MatTableModule,
+  MatTabsModule,
+  MatChipsModule,
+  MatDialog,
+  MatDialogRef,
+  MatPaginator,
+  MatSnackBar,
+  MAT_DIALOG_DATA,
+} from '@angular/material';
+import {MatFormFieldModule} from '@angular/material/form-field'; 
+import {MatSelectModule} from '@angular/material'; 
+import {RMM} from '../rmm';
+@Component({
+    selector: 'aliases-edit',
+    styles: [`
+    .mat_header {
+        padding: 10px 0 10px 10px
+    }
+    .mat_card.update .mat_header,
+    .mat_card.create .mat_header {
+        background-color: #013b69;
+     }
+    .header-image {
+       border-radius: 50%;
+       flex-shrink: 0;
+       background-image:url(/_img/avatar.svg);
+       background-size: cover;
+    }
+    `],
+    template: `
+        <mat-card class="mat_card {{css_class()}}" style="padding: 0px">
```

Comment:
`css_class()` may be called at every render frame here (in the worst case: see [here](https://medium.com/showpad-engineering/why-you-should-never-use-function-calls-in-angular-template-expressions-e1a50f9c0496)). 

It would be better to have this set statically on object creation. From what I can see in `profiles/profiles.component.ts`, these `is_whatever` settings are set exactly once on object creation. Since this is a dialog, perhaps it'd be better to pass the "mode" (maybe as an enum, `EditorMode::Create/Update/Delete`) with the `data` and then set the relevant CSS classes just once, in the constructor? Then we don't have to look it up every frame.

Reply:
resolved